File: Assignment/GroupAssignment/frontend/src/types/dto/promotion.dto.ts
import { MenuItemDTO } from './menu-item.dto';

export enum PromotionType {
  MENU_ITEM = 'MENU_ITEM',
  ORDER = 'ORDER',
}

export enum DiscountType {
  PERCENTAGE = 'PERCENTAGE',
  FIXED_AMOUNT = 'FIXED_AMOUNT',
}

export enum PromotionStatus {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE',
  EXPIRED = 'EXPIRED',
  DELETED = 'DELETED',
}

export interface PromotionDTO {
  promotionId: string;
  name: string;
  description: string;
  code: string;
  promotionType: PromotionType;
  discountType: DiscountType;
  discountValue: number;
  minOrderValue?: number;
  maxDiscountValue?: number;
  startDate: string;
  endDate: string;
  status: PromotionStatus;
  menuItems: MenuItemDTO[];
}

export interface CreatePromotionRequest {
  name: string;
  description: string;
  code: string;
  promotionType: PromotionType;
  discountType: DiscountType;
  discountValue: number;
  minOrderValue?: number;
  maxDiscountValue?: number;
  startDate: string;
  endDate: string;
  restaurantId: string;
  menuItemIds?: string[];
}

File: Assignment/GroupAssignment/backend/src/main/java/com/example/backend/dto/PromotionDTO.java
package com.example.backend.dto;

import com.example.backend.entities.DiscountType;
import com.example.backend.entities.PromotionStatus;
import com.example.backend.entities.PromotionType;
import lombok.*;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.Set;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PromotionDTO {
    private UUID promotionId;
    private String name;
    private String description;
    private String code;
    private PromotionType promotionType;
    private DiscountType discountType;
    private BigDecimal discountValue;
    private BigDecimal minOrderValue;
    private BigDecimal maxDiscountValue;
    private Instant startDate;
    private Instant endDate;
    private PromotionStatus status;
    private Set<MenuItemDTO> menuItems;
}
